contracts/liquidator/Liquidator.sol

```diff
@@ -0,0 +1,206 @@
+
+        uint256 fee = decoded.reversedPair? fee0 : fee1;
+        payback(decoded.amount, fee, comet.baseToken(), totalAmountOut, decoded.payer);
+    }
+
+    function payback(
+        uint256 amount,
+        uint256 fee,
+        address token,
+        uint256 amountOut,
+        address payer
+    ) internal {
+        uint256 amountOwed = LowGasSafeMath.add(amount, fee);
+        TransferHelper.safeApprove(token, address(this), amountOwed);
+
+        if (amountOwed > 0) pay(token, address(this), msg.sender, amountOwed);
+
+        // if profitable, pay profits to payer
+        if (amountOut > amountOwed) {
+            uint256 profit = LowGasSafeMath.sub(amountOut, amountOwed);
+            TransferHelper.safeApprove(token, address(this), profit);
+            pay(token, address(this), payer, profit);
+        }
+    }
+
+    function calculateTotalBaseAmount() internal view returns (uint256, uint256[] memory, address[] memory) {
+        uint256 totalBaseAmount = 0;
+        uint8 numAssets = comet.numAssets();
+        uint256[] memory assetBaseAmounts = new uint256[](numAssets);
+        address[] memory cometAssets = new address[](numAssets);
+        for (uint8 i = 0; i < numAssets; i++) {
+            address asset = comet.getAssetInfo(i).asset;
+            cometAssets[i] = asset;
+            uint256 collateralBalance = comet.collateralBalanceOf(address(comet), asset);
+            uint256 quotePrice = comet.quoteCollateral(asset, 1 * comet.baseScale());
+
+            /*
+                quoteCollateral = amount of DAI you get for 1 USDC
+                collateralBalance = Comet's balance of DAI
+                price = amount of USDC required to buy all DAI
+
+                1 / quotePrice = x / collateralBalance
+                (1 / quotePrice) * collateralBalance = x
+            */
+            // uint256 assetBaseAmount = comet.collateralBalanceOf(address(comet), asset) * quotePrice / 1e30; // PRICE_SCALE + 1e12
+            // console.log("assetBaseAmount: %s", assetBaseAmount);
+            uint256 assetBaseAmount = ((1e6 * 1e18 / quotePrice) * collateralBalance) / 1e18;
+            assetBaseAmounts[i] = assetBaseAmount;
+            totalBaseAmount += assetBaseAmount;
+        }
+
+        return (totalBaseAmount, assetBaseAmounts, cometAssets);
+    }
+
+    /// @param params The parameters necessary for flash and the callback, passed in as FlashParams
+    /// @notice Calls the pools flash function with data needed in `uniswapV3FlashCallback`
+    function initFlash(FlashParams memory params) external {
+        // Absorb Comet underwater accounts
+        comet.absorb(address(this), params.accounts);
+
+        (uint256 totalBaseAmount, uint256[] memory assetBaseAmounts, address[] memory cometAssets) = calculateTotalBaseAmount();
+
+        address poolToken0 = params.reversedPair ? comet.baseToken(): params.pairToken;
+        address poolToken1 = params.reversedPair ? params.pairToken : comet.baseToken();
```

Comment:
Feels a bit brittle for the client to pass in `reversedPair`. I took a look at the `PoolAddress` library and looks like we can just use [this logic](https://github.com/Uniswap/v3-periphery/blob/main/contracts/libraries/PoolAddress.sol#L20-L27) to create the `PoolKey`. You can also probably derive the value of `reversedPair` using the same logic (e.g. `if (pairToken > baseToken) reversedPair = true;`).

Reply:
Yeah token order should be reliable

Reply:
Interesting point, I will try it out.

Reply:
I added it in the latest merged pr